snowfall/training/mmi_mbr_graph.py

```diff
@@ -134,9 +134,10 @@ def compile(self, texts: Iterable[str],
         assert P.device == self.device
         P_with_self_loops = k2.add_epsilon_self_loops(P)
 
-        ctc_topo_P = k2.intersect(self.ctc_topo_inv,
-                                  P_with_self_loops,
-                                  treat_epsilons_specially=False).invert()
+        ctc_topo_P = k2.compose(self.ctc_topo,
```

Comment:
> I don't see, in the current code at least in my branch I'm working on, anywhere where the 'phones' attribute is set in den_graph.

@danpovey 

It is set here.

Reply:
The latest k2 is needed.
https://github.com/k2-fsa/k2/pull/670 prevent overwriting `num_graph`'s phones attribute.

Reply:
Looking into it, I suspect something was reversed somewhere.